WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleTransactionSuccessView.swift

```diff
@@ -22,24 +21,30 @@ struct PointOfSaleCardPresentPaymentSuccessMessageView: View {
                     VStack(alignment: .center, spacing: Constants.headerSpacing) {
                         successIcon
                             .renderedIf(!dynamicTypeSize.isAccessibilitySize)
-                            .matchedGeometryEffect(id: animation.iconTransitionId, in: animation.namespace, properties: .position)
+                            .scaleEffect(isViewLoaded ? 1 : 0)
+                            .opacity(isViewLoaded ? 1 : 0)
+
                         VStack(alignment: .center, spacing: Constants.textSpacing) {
                             Text(viewModel.title)
                                 .font(.posTitleEmphasized)
                                 .foregroundStyle(Color.posPrimaryText)
                                 .accessibilityAddTraits(.isHeader)
-                                .matchedGeometryEffect(id: animation.titleTransitionId, in: animation.namespace, properties: .position)
+                                .offset(y: isViewLoaded ? 0 : Constants.animationOffset)
```

Comment:
I think you got the necessary changes perfectly 👍 

This view doesn't animate independently.`.matchedGeometryEffect` only transitions from previous positions of the previous view. If there are no previous positions - nothing animated.

Ideally, we could have either:
a) `.matchedGeometryEffect`
a) If it doesn't exist, then manual `.offset` calculation

I would leave scaleEffect and opacity for both versions. I think it looks better 👍 But you can check with design.

Reply:
Thanks for the review and your comments in Slack! 

> Ideally, we could have either:
a) .matchedGeometryEffect
a) If it doesn't exist, then manual .offset calculation

That sounds good yeah, I found this to be quite tricky as when debugging the animation we could see the actual values for animations being passed into the view, but they would not trigger the animation, so we couldn't do a nil check here to see if they exists or not 😭 . I've also experimented a bit with capturing their values by making a `PreferenceKeys` extension that would capture them on `.onPreferenceChange` to debug/modify these, but I got nowhere.

I'll investigate a bit to see what alternatives we have in these cases 👍